pfioh/pfioh.py

```diff
@@ -818,63 +804,56 @@ def do_POST_withCompression(self, **kwargs):
-                with open(str_localFile, 'wb') as fh:
-                    fh.write(data)
+                data                        = base64.b64decode(fileContent)
                 d_ret['decode']['status']   = True
                 d_ret['decode']['msg']      = 'base64 decode successful!'
-            except:
+            except Exception as err:
                 d_ret['decode']['status']   = False
                 d_ret['decode']['msg']      = 'base64 decode unsuccessful!'
                 self.ret_client(d_ret)
                 self.qprint(d_ret, comms = 'tx')
+                self.qprint(err)
                 return d_ret
-            d_ret['decode']['timestamp']  = '%s' % datetime.datetime.now()
+
         else:
             d_ret['write']   = {}
-            with open(str_localFile, 'wb') as fh:
-                try:
-                    fh.write(fileContent.decode())
-                    d_ret['write']['decode'] = True
-                except:
-                    fh.write(fileContent)
-                    d_ret['write']['decode'] = False
-            d_ret['write']['file']      = str_localFile
-            d_ret['write']['status']    = True
-            d_ret['write']['msg']       = 'File written successfully!'
-            d_ret['write']['filesize']  = "{:,}".format(os.stat(str_localFile).st_size)
-            d_ret['write']['timestamp'] = '%s' % datetime.datetime.now()
-            d_ret['status']             = True
-            d_ret['msg']                = d_ret['write']['msg']
-        fh.close()
-        if b_unpack and d_compress['archive'] == 'zip':
-            d_fio   =   zip_process(action          = 'unzip',
-                                    path            = str_unpackPath,
-                                    payloadFile     = str_localFile)
-            d_ret['unzip']  = d_fio
-            d_ret['status'] = d_fio['status']
-            d_ret['msg']    = d_fio['msg']
-            os.remove(str_localFile)
-
+            try:
+                data = fileContent.decode()
+                d_ret['write']['decode'] = True
+            except Exception as err:
+                d_ret['write']['decode'] = False
+                data = fileContent
+                self.qprint(err)
+
+        b_zip= b_unpack and (d_compress['archive'] == 'zip')
+
+        d_ret= self.storeData(file_name= str_localFile, 
+                              file_content= data, Path= str_unpackPath, is_zip=b_zip,d_ret=d_ret)
+        
         # pudb.set_trace()
         d_ret['postop'] = self.postop_process(meta          = d_meta,
                                               path          = str_unpackPath,
                                               payloadFile   = str_localFile)
 
+
         self.send_response(200)
         self.end_headers()
 
         d_ret['User-agent'] = self.headers['user-agent']
 
         # self.ret_client(d_ret)
         self.qprint(d_ret, comms = 'tx')
-
         return d_ret
 
+    def storeData(self, **kwargs):
```

Comment:
No, this is just in case that some base class has not implemented the method, we should come to know that the method has to be implemented. Java throws an error if the base class has not implemented the method, this is kind of doing the same.  
When it does Dynamic method dispatch, it will first search for the method in the class whose object I have used to call the method with. If it does not find it there, it will jump to parent class to look for that method.